plan/stats.go

```diff
@@ -174,7 +175,23 @@ func (p *LogicalSelection) deriveStats() (*statsInfo, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	p.stats = childProfile.scale(selectionFactor)
+	factor := selectionFactor
+	if p.ctx.GetSessionVars().OptimizerSelectivityLevel >= 1 && childProfile.histColl != nil {
```

Comment:
what does `OptimizerSelectivityLevel >= 1` mean?

Reply:
Just its literal meaning?
This variable controls how we maintain stats to calculate selectivity. The higher it is, the more powerful way we use. Currently, it should be 0 or 1.

Reply:
How to support `OptimizerSelectivityLevel=2`?

Reply:
For example, when it's 1, we only use the `DataSource`'s `Histogram` and won't create newly one. When it's 2, we'll create new `Histogram` during planning.

Reply:
how about making some constant values to represent different levels, for example:
```go
const (
	SLPseudo int = 0
	SLDataSourceOnlyStats = 1
	SLAllStats = 2
)
```